### ISSUE TITLE
Multiple units flood the broker

### ISSUE THREAD

**Issue by Sprinterfreak** [gysmo38/mitsubishi2MQTT#188]
Since I was happy with my first Mitsubishi unit tamed, I wanted to fit the second.

TL;DR after flashing the second ESP32, connecting, both ESP's start to flood LWT online messages to the broker.
Perhaps not using unique Client ID's to broker?

Long Version
Hardware ESP32
Software 2022.11.0 (built with pio on ubuntu)

Both ESP's have different names below settings/wifi Hostname: HVACabc / HVACdef
Both ESP's have different names below settings/mqtt Friendy Name: HVACabc / HVACdef
Both using default topic: mitsubishi2mqtt
Happens with and without mqtt authentication

Both ESP's are located in the same L2 network as the broker and get unique IP's via DHCP.

I'm using Homeassistant.

**Comment by Sprinterfreak** [comment]
I'm using ESP32 modules from az-delivery.
All ESP modules from az-delivery have the same Chip ID ef49c0 which causes getID() to be not a unique id.

Intrestingly does the mac appear to be unique.

**Comment by Sprinterfreak** [comment]
Turned out that getId() returned the wrong half of the MAC as unique identifier. az-delivery is not to blame here.
The vendor part of the mac is to be expected static on a batch of ESP's ;)

Using the serial part of the MAC from EfuseMac seem to have solved this issue.
